src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2023 Dynatrace LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This file includes a Java port of the Imohash algorithm originally published
+ * at https://github.com/kalafut/imohash under the following license:
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Jim Kalafut
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.dynatrace.hash4j.file;
+
+import static com.dynatrace.hash4j.util.Preconditions.checkArgument;
+
+import com.dynatrace.hash4j.hashing.HashStream128;
+import com.dynatrace.hash4j.hashing.HashValue128;
+import com.dynatrace.hash4j.hashing.Hashing;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+
+class Imohash1_0_2 extends AbstractFileHasher128 {
+
+  static final long DEFAULT_SAMPLE_THRESHOLD = 128 * 1024;
+  static final int DEFAULT_SAMPLE_SIZE = 16 * 1024;
+
+  private static final int BUFFER_SIZE = 1024;
```

Comment:
I'd increase the buffer size to at least 4KB, as thats the common default block size for ntfs and ext4.

Reply:
changed buffer size to 4kB